citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationVersionManager.java

```diff
@@ -0,0 +1,70 @@
+package edu.asu.diging.citesphere.core.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.citesphere.core.bib.ICitationVersionsDao;
+import edu.asu.diging.citesphere.core.exceptions.AccessForbiddenException;
+import edu.asu.diging.citesphere.core.exceptions.GroupDoesNotExistException;
+import edu.asu.diging.citesphere.core.model.bib.CitationVersion;
+import edu.asu.diging.citesphere.core.service.ICitationVersionManager;
+import edu.asu.diging.citesphere.data.bib.CitationGroupRepository;
+import edu.asu.diging.citesphere.model.bib.ICitation;
+import edu.asu.diging.citesphere.model.bib.ICitationGroup;
+import edu.asu.diging.citesphere.user.IUser;
+
+@Service
+public class CitationVersionManager implements ICitationVersionManager {
+
+    // FIXME: needs to be replaced with group manager
```

Comment:
what is this comment about?

Reply:
I copied it over from CitationManager. I figured the comment would apply here as well.

Reply:
Can you use group manager instead here?

Reply:
Yes. I was just looking into it and I'll make the change.

Reply:
As of now, `getGroup()` method of GroupManager does not work properly. It returns a group as long as there is some group with the given id even if the user does not belong to it. It also creates a local entry for it, adding the user to it. Should I fix it now or would you create a separate ticket for this?

Reply:
Are you talking about this method: https://github.com/diging/citesphere/blob/develop/citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/GroupManager.java#L42

That only seems to return a group if the user has access to it.

Reply:
Yes. I tried random groupIds and it provided a response.

Reply:
what do you mean by "random groupIds"?

Reply:
![image](https://user-images.githubusercontent.com/25862427/129788553-de185b91-1547-4cc1-93dc-1ec0fbd41968.png)

Reply:
Does it provide a response as long as the user has access to it and not necessarily be a part of it?

Reply:
That is actually not wrong. It looks like that's a public group, so you get it from Zotero. The method does what it is supposed to do. And you just need it to check if a user has access, right? So you should be able to use the method (we just have to implement better permission checks on those groups eventually).